dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java

```diff
@@ -0,0 +1,124 @@
+
+    public static final String SERVER_NAME = "weldDevModeServer";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    //Does not repeat against CDI 4.0 where dev mode is not supported
+    @ClassRule
+    public static RepeatTests r = CDIExtensionRepeatActions.repeat(SERVER_NAME, CDIExtensionRepeatActions.EE7_PLUS, CDIExtensionRepeatActions.EE9_PLUS);
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        System.out.println("Install the user feature bundle... cdi.helloworld.extension");
+        CDIExtensionRepeatActions.installUserExtension(server, CDIExtensionRepeatActions.HELLOWORLD_EXTENSION_BUNDLE_ID);
+        CDIExtensionRepeatActions.installSystemFeature(server, CDIExtensionRepeatActions.CDI_INTERNALS_BUNDLE_ID);
+
+        JavaArchive multipleWarEmbeddedJar = ShrinkWrap.create(JavaArchive.class, "multipleWarEmbeddedJar.jar");
+        multipleWarEmbeddedJar.addClass(EmbeddedJarMyEjb.class);
+        CDIArchiveHelper.addBeansXML(multipleWarEmbeddedJar, DiscoveryMode.ANNOTATED);
+
+        WebArchive multipleWarNoBeans = ShrinkWrap.create(WebArchive.class, "multipleWarNoBeans.war");
+        multipleWarNoBeans.addClass(NoBeansTestServlet.class);
+
+        WebArchive multipleWar = ShrinkWrap.create(WebArchive.class, "multipleWar1.war");
+        multipleWar.addClass(WAR1TestServlet.class);
+        multipleWar.addClass(WAR1MyBean.class);
+        multipleWar.addAsWebInfResource(WAR1TestServlet.class.getPackage(), "ejb-jar.xml", "ejb-jar.xml");
+        CDIArchiveHelper.addBeansXML(multipleWar, DiscoveryMode.ANNOTATED);
+        multipleWar.addAsLibrary(multipleWarEmbeddedJar);
+
+        EnterpriseArchive multipleWars = ShrinkWrap.create(EnterpriseArchive.class, "multipleWars2.ear");
+        multipleWars.setApplicationXML(NoBeansTestServlet.class.getPackage(), "application.xml");
+        multipleWars.addAsModule(multipleWar);
+        multipleWars.addAsModule(multipleWarNoBeans);
+
+        ShrinkHelper.exportDropinAppToServer(server, multipleWars, DeployOptions.SERVER_ONLY);
+
+        server.startServer(true);
+    }
+
+    @Test
+    public void testExtensionLoaded() throws Exception {
+        assertStringsInLogs("WELD-000411: Observer method \\[BackedAnnotatedMethod\\] public org.jboss.weld.probe.ProbeExtension.processAnnotatedTypes");
+        assertStringsInLogs("Weld Development Mode: ENABLED");
+
+        assertProbeProcessedBDA("multipleWars2#org.jboss.weld.probe.ProbeExtension.additionalClasses");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war#WEB_INF_LIB#WEB-INF/lib/multipleWarEmbeddedJar.jar");
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.beanvalidation.v11.cdi_1.additionalClasses"); //BDA name is very different in EE9
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.jsf.2.2_1.additionalClasses");                //BDA name is very different in EE9
```

Comment:
Remove commented out code Could add a new comment that we don't check for the BDAs added by BV or JSF since the name varies by version. Alternatively you could assert that one of the expected BDAs is present.

Reply:
Since we don't really need to check for ALL probe BDAs, it is easier to just ignore these ones